learn/guides/calling-java-code-from-ballerina.md

```diff
@@ -0,0 +1,290 @@
+---
+layout: ballerina-calling-java-code-left-nav-pages-swanlake
+title: Calling Java Code from Ballerina
+description: See how Ballerina offers a straightforward way to call existing Java code from Ballerina.
+keywords: ballerina, programming language, java api, interoperability
+permalink: /learn/calling-java-code-from-ballerina/
+active: calling-java-code-from-ballerina
+intro: Ballerina offers a straightforward way to call the existing Java code from Ballerina. Although Ballerina is not designed to be a JVM language, the current implementation, which targets the JVM, aka jBallerina, provides Java interoperability by adhering to the Ballerina language semantics.
+redirect_from:
+  - /learn/how-to-use-java-interoperability
+  - /learn/how-to-use-java-interoperability/
+  - /learn/how-to-call-java-code-from-ballerina
+  - /learn/how-to-call-java-code-from-ballerina/
+  - /learn/calling-java-code-from-ballerina
+  - /swan-lake/learn/calling-java-code-from-ballerina
+  - /swan-lake/learn/calling-java-code-from-ballerina/
+  - /learn/user-guide/calling-java-code-from-ballerina
+  - /learn/user-guide/calling-java-code-from-ballerina/
+  - /learn/user-guide/interoperability/calling-java-code-from-ballerina
+  - /learn/user-guide/interoperability/calling-java-code-from-ballerina/
+  - /learn/user-guide/interoperability/
+  - /learn/user-guide/interoperability
+  - /learn/calling-java-code-from-ballerina-and-vice-versa
+  - /learn/calling-java-code-from-ballerina-and-vice-versa/
+---
+
+## Writing Ballerina Bindings
+We have introduced a CLI tool called `bindgen` to ease the process of generating Ballerina bindings for given Java APIs. This guide shows you how to use it, if you need more details on how the bindgen tool works, you can refer its reference guide [here](/learn/java-interoperability/the-bindgen-tool/).
+
+The [Package Layout Guide](/learn/organizing-ballerina-code/package-layout/) explains how to package Java libraries (JAR files) with Ballerina programs. When you generate bindings for a Java library using the bindgen tool, this part is already handled. However, you could gain an in-depth understanding of how Java libraries and their transitive dependencies are packaged with Ballerina programs to produce self-contained executable programs.
+
+The [Ballerina FFI Guide](/learn/java-interoperability/ballerina-ffi/) explains how to write these bindings manually. It is also a useful section for those who want to understand the inner workings of calling Java from Ballerina and for those who want to customize the bindings generated by the bindgen tool.
+
+## Using the SnakeYAML Java Library in Ballerina
+SnakeYAML is a YAML parser for Java. In this section, we'll learn how to use this library to parse a YAML document using Ballerina. 
```

Comment:
```suggestion
SnakeYAML is a YAML parser for Java. In this section, you will learn how to use this library to parse a YAML document using Ballerina. 
```
Please change all occurrences of `we` to `you`.

Reply:
Done.